File: java/src/main/java/de/xjustiz/core/models/Telekommunikationsart.java
package de.xjustiz.core.models;

import com.fasterxml.jackson.annotation.JsonAlias;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
import jakarta.xml.bind.annotation.XmlAccessType;
import jakarta.xml.bind.annotation.XmlAccessorType;
import jakarta.xml.bind.annotation.XmlAttribute;
import jakarta.xml.bind.annotation.XmlElement;
import jakarta.xml.bind.annotation.XmlTransient;
import org.jetbrains.annotations.Nullable;

/**
 * Die Art der Telekommunikation.
 * <p>
 * <u><b>Telecommunication type:</b></u> The type of telecommunication.
 */
@XmlAccessorType(XmlAccessType.FIELD)
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_EMPTY)
public class Telekommunikationsart {

    /**
     * Die Art der Telekommunikation.
     * <p>
     * <u><b>Telecommunication type:</b></u> The type of telecommunication.
     */
    @XmlAttribute(name = "listVersionID")
    @JacksonXmlProperty(isAttribute = true, localName = "listVersionID")
    @JsonProperty("ListVersionId")
    private String listVersionId;

    /**
     * Die Art der Telekommunikation.
     * <p>
     * <u><b>Telecommunication type:</b></u> The type of telecommunication.
     */
    @XmlAttribute(name = "listURI")
    @JacksonXmlProperty(isAttribute = true, localName = "listURI")
    @JsonProperty("ListUri")
    @Nullable
    private String listUri;

    /**
     * Ruft den Telekommunikationsart-Code ab oder legt diesen fest.
     * <p>
     * <u><b>Code:</b></u> Gets or sets the telecommunication type code.
     */
    @XmlTransient
    @JsonProperty("code")
    @JsonAlias({ "Code", "code" })
    private TelekommunikationsartCode code;

    /**
     * Hilfseigenschaft für die XML-Serialisierung.
     * <p>
     * <u><b>Code for XML:</b></u> Helper property for XML serialization.
     */
    @XmlElement(name = "code", namespace = "http://www.xjustizcore.de")
    @JacksonXmlProperty(localName = "code", namespace = "http://www.xjustizcore.de")
    @JsonIgnore
    private String codeForXml;

    public Telekommunikationsart() {}

    public String getListVersionId() { return listVersionId; }
    public void setListVersionId(String listVersionId) { this.listVersionId = listVersionId; }

    public String getListUri() { return listUri; }
    public void setListUri(String listUri) { this.listUri = listUri; }

    public TelekommunikationsartCode getCode() { return code; }
    public void setCode(TelekommunikationsartCode code) { this.code = code; }

    public String getCodeForXml() { return codeForXml; }
    public void setCodeForXml(String codeForXml) { this.codeForXml = codeForXml; }

}

File: sdk/dotnet/src/xjustiz.core-dotnet/Models/Entities/Telekommunikationsart.cs
namespace xjustiz.core_dotnet.Models.Entities;

using System.Text.Json.Serialization;
using System.Xml.Serialization;
using xjustiz.core_dotnet.Models.Codes;
using xjustiz.core_dotnet.Models.Codes.Telekommunikationsart;
using xjustiz.core_dotnet.Models.Helpers;
using xjustiz.core_dotnet.Util.Versioning;

/// <summary>
/// Die Art der Telekommunikation.<br/>
/// <u><b>Telecommunication type:</b></u> The type of telecommunication.
/// </summary>
[XJustizAvailability(XJustizVersion.V2_1_0)]
[XJustizCoreAvailability(XJustizCoreVersion.V0_2_0)]
public class Telekommunikationsart : ICode<TelekommunikationsartCode>
{
    /// <inheritdoc/>
    [XmlAttribute("listVersionID")]
    public string ListVersionId { get; set; } = TelekommunikationsartCodeLists.LatestList.Version;

    /// <inheritdoc/>
    [XmlAttribute("listURI")]
    public string? ListUri { get; set; } = TelekommunikationsartCodeLists.Uri;

    /// <summary>
    /// Ruft den Telekommunikationsart-Code ab oder legt diesen fest.<br/>
    /// <u><b>Code:</b></u> Gets or sets the telecommunication type code.
    /// </summary>
    [XmlIgnore]
    [JsonPropertyName("code")]
    public TelekommunikationsartCode Code { get; set; }

    /// <summary>
    /// Hilfseigenschaft für die XML-Serialisierung.<br/>
    /// <u><b>Code for XML:</b></u> Helper property for XML serialization.
    /// </summary>
    [XmlElement("code", Namespace = SchemeConstants.XJustizCore_Tns)]
    [JsonIgnore]
    public string CodeForXml
    {
        get => Code.ToCode();
        set
        {
            if (TelekommunikationsartCodeMapper.TryParse(value, out var result))
            {
                Code = result;
            }
            else
            {
                Code = default;
            }
        }
    }
}
